### PR TITLE
Add an avatar view to easily show BoxAvatars.

### PR DESCRIPTION
**Pull request by doncung**

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @doncung has signed the CLA. Thanks for the pull request!

File: box-content-sdk/src/main/java/com/box/androidsdk/content/views/BoxAvatarView.java [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+
+        if (avatarFile.exists()){
+
+            // load avatar file into view.
+            mAvatar.setImageDrawable(Drawable.createFromPath(avatarFile.getAbsolutePath()));
+            mAvatar.setVisibility(View.VISIBLE);
+            mInitials.setVisibility(View.GONE);
+        } else {
+
+            String name = "";
```
Comment (seema-at-box) [inline box-content-sdk/src/main/java/com/box/androidsdk/content/views/BoxAvatarView.java]: Move to a constant / initialize as null instead.

File: box-content-sdk/src/main/java/com/box/androidsdk/content/views/DefaultAvatarController.java [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    protected transient ThreadPoolExecutor mExecutor;
+
+    public DefaultAvatarController(BoxSession session) {
+        mSession = session;
+        mApiUser = new BoxApiUser(session);
+    }
+
+
+    protected File getAvatarDir(final String userId) {
+        File avatarDir = new File(mSession.getCacheDir(), "avatar");
```
Comment (seema-at-box) [inline box-content-sdk/src/main/java/com/box/androidsdk/content/views/DefaultAvatarController.java]: Move to a constant

File: box-content-sdk/src/main/java/com/box/androidsdk/content/views/DefaultAvatarController.java [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+
+/**
+ * This view is used to view avatars.
+ */
+public class DefaultAvatarController implements BoxAvatarView.AvatarController, Serializable {
+
+    protected BoxSession mSession;
+    protected transient BoxApiUser mApiUser;
+
+    protected HashSet<String> mUnavailableAvatars = new HashSet<String>();
```
Comment (seema-at-box) [inline box-content-sdk/src/main/java/com/box/androidsdk/content/views/DefaultAvatarController.java]: When is this recyled/okay for trying again?

File: box-content-sdk/src/main/java/com/box/androidsdk/content/views/BoxAvatarView.java [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+
+    }
+
+
+    public <T extends Serializable & AvatarController> void loadUser(final BoxCollaborator collaborator, T avatarController){
+        if (avatarController != null){
+            mAvatarController = avatarController;
+        }
+        if (mUser != null && collaborator != null && mUser.getId().equals(collaborator.getId())){
+            // if this is called with the same user do nothing.
```
Comment (doncung) [inline box-content-sdk/src/main/java/com/box/androidsdk/content/views/BoxAvatarView.java]: If the user is the same we know that we have already loaded this user.

File: box-content-sdk/src/main/java/com/box/androidsdk/content/views/DefaultAvatarController.java (hunk 3 above)
Comment (doncung) [inline box-content-sdk/src/main/java/com/box/androidsdk/content/views/DefaultAvatarController.java]: When the hashset is removed from memory.

**Comment by seema-at-box** [comment]
+1

**Pull request merged**